src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -280,11 +282,29 @@ private int tickAllCapacitors() {
         int supply = 0;
 
         for (Map.Entry<Location, EnergyNetComponent> entry : capacitors.entrySet()) {
-            supply += entry.getValue().getCharge(entry.getKey());
+            supply = flowSafeAddition(supply, entry.getValue().getCharge(entry.getKey()));
         }
 
         return supply;
     }
+    
+    /**
+     * This detects if 2 integers will overflow/underflow and if they will, returns the corresponding value
+     * @param i1 the first integer
+     * @param i2 the second integer
+     * @return {@link Integer#MAX_VALUE} if overflow detected, {@link Integer#MIN_VALUE} if underflow detected, otherwise the sum of i1 and i2
+     */
+    private int flowSafeAddition(int i1, int i2) {
```

Comment:
I think `a` and `b` would be better to distinguish the two params.

Reply:
Sure! Wasn't sure which direction I wanted to go 😅